File: html/redcap_v15.7.6/API/examples/r/export_file.R
#!/usr/bin/env Rscript

source('config.R')
library(RCurl)

result <- postForm(
    api_url,
    token=api_token,
    content='file',
    action='export',
    record='f21a3ffd37fc0b3c',
    field='file_upload',
    event='event_1_arm_1'
)
print(result)

File: redcap_v15.7.6/API/examples/ruby/export_file.rb
#!/usr/bin/env ruby

require 'curl'
require './settings.rb'
include Settings

fields = {
  :token => Settings::API_TOKEN,
  :content => 'file',
  :action => 'export',
  :record => 'f21a3ffd37fc0b3c',
  :field => 'file_upload',
  :event => 'event_1_arm_1'
}

ch = Curl::Easy.http_post(
  Settings::API_URL,
  fields.collect{|k, v| Curl::PostField.content(k.to_s, v)}
)
puts ch.body_str
